netdata/netdata
# Fix node information send to the cloud for older agent versions

Fixes https://github.com/netdata/netdata-cloud/issues/293
##### Summary
In a child - parent setup when an child is running a very old agent version it doesn't send information that the
parent running a newer version expects. 

If the parent is claimed and connected to the cloud, it will attempt to build a node info payload and it will crash.

This PR adds checks for 
 -  host cores
 - CPU frequency
 - Total RAM
 - Disk space
 - Virtualization
 - Container
- Fixes an issue with transmitting the wrong agent version to the cloud for children connecting



##### Test Plan
- Setup parent - child 
  - Parent should run the latest version
  - Child should run an older version (try v1.19.0)
- Connect parent to the cloud
   - Notice the parent that crashes
- Apply the PR and try again
